Assets/Mirror/Examples/MultipleAdditiveScenes/Scripts/RandomColor.cs

```diff
@@ -0,0 +1,32 @@
+using UnityEngine;
+
+namespace Mirror.Examples.MultipleAdditiveScenes
+{
+    public class RandomColor : NetworkBehaviour
+    {
+        public override void OnStartServer()
+        {
+            base.OnStartServer();
+            color = Random.ColorHSV(0f, 1f, 1f, 1f, 0.5f, 1f);
+        }
+
+        // Color32 packs to 4 bytes
+        [SyncVar(hook = nameof(SetColor))]
+        public Color32 color = Color.black;
+
+        // Unity clones the material when GetComponent<Renderer>().material is called
+        // Cache it here and destroy it in OnDestroy to prevent a memory leak
+        Material cachedMaterial;
+
+        void SetColor(Color32 _, Color32 newColor)
+        {
+            if (cachedMaterial == null) cachedMaterial = GetComponentInChildren<Renderer>().material;
```

Comment:
move this caching  check to a property

Reply:
Separate PR since RandomColor is copied to 3 examples.